File: 1-two-sum/solution1.cpp
#include <vector>

class Solution
{
public:
    std::vector<int> twoSum(std::vector<int> &nums, int target)
    {
        std::vector<int> results(2);
        int numsSize = nums.size();

        for (int i = 0; i < numsSize - 1; i++)
        {
            for (int j = i + 1; j < numsSize; j++)
            {
                if (nums[i] + nums[j] == target)
                {
                    results[0] = i;
                    results[1] = j;
                    return results;
                }
            }
        }

        return results;
    }
};

int main()
{
    using namespace std;

    vector<int> nums = { 2, 7, 11, 15 };
    int target = 22;
}

File: 1-two-sum/solution1.cs
using System;

public class Solution
{
    public static void Main(String[] args)
    {
        Int32[] nums = new Int32[] { 2, 7, 11, 15 };
        Int32 target = 22;
        
        try
        {
            Int32[] results = TwoSum(nums, target);

            foreach (var item in results)
            {
                Console.Write(item + " "); // 1 3
            }
        }
        catch (System.Exception ex)
        {
            Console.WriteLine(ex);
        }

    }

    public static Int32[] TwoSum(Int32[] nums, Int32 target)
    {
        for (Int32 i = 0; i < nums.Length - 1; i++)
        {
            for (Int32 j = i + 1; j < nums.Length; j++)
            {
                if (nums[i] + nums[j] == target)
                {
                    return new Int32[] { i, j };
                }
            }
        }

        throw new Exception("No two sum solution");
    }
}

// Compile: csc solution1.cs
// Run: solution1.exe